File: summer/WebContent/js/system/windows/FullValueSource.js
define(["dojo/_base/declare"], function(declare){

    var FullValueSource=declare("FullValueSource", null,{ 
    });	
    
    // Bit used to store BaseValueSourceInternal = 0x01 
    // Bit used to store BaseValueSourceInternal = 0x02
    // Bit used to store BaseValueSourceInternal = 0x04
    // Bit used to store BaseValueSourceInternal = 0x08
 
	FullValueSource.ValueSourceMask     = 0x000F;
	FullValueSource.ModifiersMask       = 0x0070; 
	FullValueSource.IsExpression        = 0x0010; 
	FullValueSource.IsAnimated          = 0x0020;
	FullValueSource.IsCoerced           = 0x0040; 
	FullValueSource.IsPotentiallyADeferredReference = 0x0080;
	FullValueSource.HasExpressionMarker = 0x0100;
	FullValueSource.IsCoercedWithCurrentValue = 0x200;

    
    return FullValueSource;
});

File: org.summer.view.widget/src/org/summer/view/widget/FullValueSource.java
package org.summer.view.widget;
//[FriendAccessAllowed] // Built into Base, also used by Core & Framework. 
/*internal*/public enum FullValueSource //: short 
{
    // Bit used to store BaseValueSourceInternal = 0x01 
    // Bit used to store BaseValueSourceInternal = 0x02
    // Bit used to store BaseValueSourceInternal = 0x04
    // Bit used to store BaseValueSourceInternal = 0x08

    ValueSourceMask    ,// = 0x000F,
    ModifiersMask      ,// = 0x0070, 
    IsExpression       ,// = 0x0010, 
    IsAnimated         ,// = 0x0020,
    IsCoerced          ,// = 0x0040, 
    IsPotentiallyADeferredReference ,//= 0x0080,
    HasExpressionMarker ,//= 0x0100,
    IsCoercedWithCurrentValue ,//= 0x200,
} 
